src/Sulu/Component/Content/Document/Syncronization/DocumentRegistrator.php

```diff
@@ -0,0 +1,214 @@
+<?php
+
+/*
+ * This file is part of Sulu.
+ *
+ * (c) MASSIVE ART WebServices GmbH
+ *
+ * This source file is subject to the MIT license that is bundled
+ * with this source code in the file LICENSE.
+ */
+
+namespace Sulu\Component\Content\Document\Syncronization;
+
+use PHPCR\Util\PathHelper;
+use Sulu\Component\Content\Document\Behavior\SynchronizeBehavior;
+use Sulu\Component\DocumentManager\Behavior\Mapping\ParentBehavior;
+use Sulu\Component\DocumentManager\DocumentManagerInterface;
+
+/**
+ * Class responsible for registering a document from the default document manager
+ * with the publish document manager including any documents dependent on
+ * the document to be synchronized.
+ */
+class DocumentRegistrator
+{
+    /**
+     * @var DocumentManagerInterface
+     */
+    private $defaultManager;
+
+    /**
+     * @var DocumentManagerInterface
+     */
+    private $publishManager;
+
+    public function __construct(
+        DocumentManagerInterface $defaultManager,
+        DocumentManagerInterface $publishManager
+    ) {
+        $this->defaultManager = $defaultManager;
+        $this->publishManager = $publishManager;
+    }
+
+    /**
+     * Register the incoming DDM document with any existing PHPCR node in the
+     * PDM.
```

Comment:
In this case it is justified writing `published document manager` and `default document manager` everywhere is confusing in the documentation. Abbreviations are expanded in the class doc.